### PR TITLE
put back the old robotino model (including hand) and removed the moveit-only model

### PR DESCRIPTION
**Pull request by shangl**

### CODE REVIEW

**Comment by ipa-nhg** [comment]
@shangl  is it really necessary add the "virtual joints" to the **robot_description** model? why? I though that jointx, jointy and jointz are only used to plan with moveit! , you move the base using the twist control /cmd_vel,  and this topic will update the position of /base_link, and the tf between /odom and /base_link.

If I add the joints to the model,  gazebo will not be able to spawn the robot, for gazebo it is a malformed model, because the linkx, linky and linkz are not proper links (no mass no inertia no control...) I would like to know if moveit is working properly with the model and if it is possible to launch the navigation with the manipulation together, before merge the model. 

anyway , this pull request doesn't work with the actual squirrel_common version:
`Failed to build tree: parent link [base_linkz] of joint [arm_base_joint] not found.  This is not valid according to the URDF spec. Every link you refer to from a joint needs to be explicitly defined in the robot description. To fix this problem you can either remove this joint [arm_base_joint] from your urdf file, or add "<link name="base_linkz" />" to your urdf file.`

**Comment by shangl** [comment]
@ipa-nhg : i just wanted to write you a mail about it. in short: yes it is. moveit cannot plan in 8-dof otherwise. i just rolled it back now to make the stuff working again (so that i finalize all the tasks assigned to me until now - as we discussed on the skype call). now planning is working in Cartesian (disabled collision checking for the hand now) and joint space (this also affects qusais kinesthetic teaching). moving the joints away from the chain destroyed the planner and the moveit model - to my knowledge there is no way out of this.

we can also discuss the exact problem via skype. however, the solution will have to be negotiated with the future maintainer (whoever that is)

edit: yes, i am aware of that. i didnt make the pull request for the other repos yet - i wasnt aware that the changes of yesterday automatically are added to this pull request. i wanted to discuss that with you before i make a pull request

**Comment by ipa-nhg** [comment]
@shangl ok,  probably moveit! needs the /joint_state of the virtual joints to be able to plan.. I will merge your pull request to `new_structure` branch , the unique consequence is that the uibk-robotino2 simulation will not work on this branch. 

just change this include https://github.com/squirrel-project/squirrel_robotino/pull/151/files#diff-b7819ce2a3e47fee9db9cb056f7e572fR7 to `<xacro:include filename="$(find robotino_description)/urdf/base_plan.urdf.xacro" />`

The future maintainer of the arm_driver/moveit have to find a solution, I can imagine that moveit have an especial configuration for this cases, or we need an extra tf publisher node.

**Comment by shangl** [comment]
@ipa-nhg : what do you mean by virtual joints? as from the view of the controller, those are real joints (and the wheels are basically kind of virtual). i really think we should discuss that (just shortly).

@Senka2112 @lokalmatador @piater @mzillich : do you have any comments or wishes about this?

**Comment by ipa-nhg** [comment]
@shangl I don't care about real or virtual joints.. 
But the kinematic is not right (from the point of view of the model!!!!!!!)  because at the moment:
-  if you move base_jointx:  base_link will not move.
-  if you move base_jointy:  base_linkx and base_link will not move.
-  if you move base_jointz:  base_linkx, base_linky and base_link will not move.

And theoretically that you want is that each joint represent an axis of base_link , and a movement of one of them update also the position of the others.

At the moment if you rotate the jointz ( the model , update the joint state ) , the jointx will not point to x direction  , the jointy will not point to y direction and base_link will stay of the (0,0,0) position (like the odometry frame) see : 
screenshot from 2016-09-13 12 04 36

Of course all of them are not real movements, you only have /cmd_vel to command , cmd_vel will update the position of base_link, but as kinematic model , from my point of view it is wrong! And make it working with gazebo is not easy.

If there are not any other solution I will merge it and, I don't know how,  but I will adapt the simulation. I just want to know that this model was properly tested, before I spend more time doing it working.

launch the nav localize the robot on a map -->  plan --> move arm +base 8dof -->  move base using the navigation --> plan --> move arm+base 8dof --> move base using the joystick --> plan --> move arm+base 8dof..... 

Check in rviz, relative position between map --> odom , odom --> base_link  and position of the arm joints before and after each movement.  

As you mention, you will not be the responsible of Moveit! , than I suggest merge this pull request to the new_structure branch , to save your changes and keep somewhere your version  and the new maintainer should take a look on this issue.

**Comment by shangl** [comment]
i dont really see why this is a conceptual problem given that there should be a 8-dof planner ( @Senka2112 , @piater : this is related to the concern i raised concerning having a 8-dof controller mixing up relative and absolute coordinates). what you show in the screenshot is just how 2-D euclidian space is parametrized (and we did this together exactly this way in the integration meeting - changes in this structure directly affect many uibk components from the low level driver up to kinesthetic teaching). the only conceptually wrong thing in the screenshot is potentially the name "base_link" - it should be something like "origin".

but again: i dont think we can clear this up on a github issue page. i am available for a skype talk or at the integration meeting

concerning testing: i tested the 8-dof planning in a test program and it works. i am not aware how to run navigation etc... and what is required to do that ( maybe @Senka2112 ? ) .

**Comment by Senka2112** [comment]
Hi
so here is on navigation:
https://github.com/squirrel-project/squirrel_nav

and I wa always starting it with the https://github.com/squirrel-project/squirrel_nav
I also never used navigation to mover robot (except from their rviz plugin) but can try to figure it out. when are these test planned? next week during inegration week?

**Comment by Senka2112** [comment]
@shangl  @ipa-nhg 
if I understand the story you two wrote well base_linkx, base_linkz, base_linky  are need to be in the model because of 8-dof planning? 
couldnt tere just additional publisher be added which is listening cmd_vel and publising transformations for  base_linkx, base_linkz, base_linky?

**Comment by shangl** [comment]
@Senka2112 : no, moevit can just not plan if the joints are not part of the model (and in particular of the same chain)

File: robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf [hunk 1]
```diff
@@ -31,6 +30,13 @@
   <!-- Wheel transmission   -->
   <!--GAZEBO PLUGIN-->
   <!-- GAZEBO OPENNI PLUGIN -->
+  <joint name="origin_joint" type="prismatic">
```
Comment (NadiaHG) [inline robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf:33]: @shangl how did you generate this file?

File: robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf.xacro [hunk 2]
```diff
@@ -20,11 +30,11 @@
     </plugin>
   </gazebo>
 
-  <hokuyo name="hokuyo" parent="base_link">
+  <hokuyo name="hokuyo" parent="base_linkz">
     <origin xyz="${hokuyo_x} ${hokuyo_y} ${hokuyo_z}" rpy="${hokuyo_roll} ${hokuyo_pitch} ${hokuyo_yaw}"/>
   </hokuyo>
 
-  <xacro:arm name="arm_" parent="base_link">
+  <xacro:arm name="arm_" parent="base_linkz">
```
Comment (NadiaHG) [inline robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf.xacro:37]: something is missed:

`Failed to build tree: parent link [base_linkz] of joint [arm_base_joint] not found. This is not valid according to the URDF spec. Every link you refer to from a joint needs to be explicitly defined in the robot description. To fix this problem you can either remove this joint [arm_base_joint] from your urdf file, or add "<link name="base_linkz" />" to your urdf file.`

see second paragraph https://github.com/squirrel-project/squirrel_robotino/pull/151#issuecomment-246616742

**Comment by ipa-nhg** [comment]
@shangl could you please summarize the result of your tests? did you have problems with the navigation? 

Just to have everything documented .

Thanks!

File: robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf (hunk 1 above)
Comment (shangl) [reply robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf:33]: with the makefile. from robotino.urdf.xacro

File: robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf.xacro (hunk 2 above)
Comment (shangl) [reply robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf.xacro:37]: should be defined in the squirrel_common repo (https://github.com/shangl/squirrel_common/commits/indigo_dev) . i checked it, it indeed is not there. obviously i forgot to push squirrel_common. will check that tomorrow

**Comment by shangl** [comment]
@ipa-nhg : 
1) created map
2) localized
3) drove around by hand with the xbox controller
4) planned stuff with the uibk controller and moveit
5) planned path using navigation
6) planned again with uibk controller
7) checked localization during all of that

@ayushais : can you confirm?

**Comment by ayushais** [comment]
Yes we did all of this and everything worked!!

**Comment by shangl** [comment]
@ipa-nhg : just one remark: new stuff should _not_ be attached to base_link but to base_linkz. otherwise 8-dof planning will break down again. we are aware that this is not a clean and optimal solution but we don't have the resources to fix that. now, navigation will have to be started oriented around base_linkz if the controller for a robot with arm is started and around base_link for a robot without an arm

@ayushais

File: robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf (hunk 1 above)
Comment (NadiaHG) [reply robotino_bringup/robots/uibk-robotino2/urdf/robotino.urdf:33]: and where is the joint between origin and base_link defined?

**Pull request closed without merging**